### PR TITLE
Add breadcrumbs to TextOverflow

### LINKED ISSUE #6910: TextOverflow docs have no breadcrumbs

### PR DESCRIPTION
**Pull request by esarbanis**
## Description

Add breadcrumbs to TextOverflow dartdoc

## Related Issues

fixes #6910

## Tests

I added the following tests:

Not tests are needed

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 1]
```diff
@@ -16,6 +16,9 @@ import 'debug.dart';
 import 'object.dart';
 
 /// How overflowing text should be handled.
+///
+/// A [TextOverflow] should be passed to a [RenderParagraph] widget via its
```
Comment (goderbauer) [inline packages/flutter/lib/src/rendering/paragraph.dart]: nit: You can also use the TextOverflow in other places. Maybe replace the "should" with "can"?

Comment (goderbauer) [inline packages/flutter/lib/src/rendering/paragraph.dart]: RenderParagraph is not a widget. Maybe just remove the word widget altogether?

Comment (goderbauer) [inline packages/flutter/lib/src/rendering/paragraph.dart]: Also: Why should we call out the RenderParagraph specifically? My guess is that this is most commonly used with the Text widget.

Comment (esarbanis) [reply packages/flutter/lib/src/rendering/paragraph.dart]: You are right it has a lot more usages, I tried to gather them all in my update

Comment (goderbauer) [reply packages/flutter/lib/src/rendering/paragraph.dart]: I would just list the most common use cases, which are Text and RichText. The other use cases are a little more esoteric and listing them seems to be more confusing than helpful.

Comment (esarbanis) [reply packages/flutter/lib/src/rendering/paragraph.dart]: OK, I will keep `Text` and `RichText` references

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**